### PR TITLE
Implement vanilla vehicle restrictions

### PR DESCRIPTION
**Pull request by GnomedDev**
Part of the fix for #667, undoes the regression made by adding the new settings menu, but does not implement further filtering due to added complexity.

### CODE REVIEW

File: payload/sp/settings/ClientSettings.hh [hunk 1]
```diff
@@ -127,10 +127,6 @@ enum class Vehicles {
     All,
     Karts,
     Bikes,
-    InsideDrift,
-    OutsideDrift,
-    Optimal,
-    Random,
```
Comment (stblr) [inline payload/sp/settings/ClientSettings.hh:133]: Please comment them out like the rest.

File: payload/game/ui/SingleTopPage.cc [hunk 2]
```diff
@@ -94,9 +118,6 @@ void SingleTopPage::onInit() {
 void SingleTopPage::onActivate() {
     m_replacement = PageId::None;
 
-    auto *context = SectionManager::Instance()->globalContext();
-    context->_74 = 2;
-
```
Comment (stblr) [inline payload/game/ui/SingleTopPage.cc:99]: What happens in time trials?

File: payload/game/ui/MultiTopPage.cc [hunk 3]
```diff
@@ -64,7 +64,7 @@ void MultiTopPage::onActivate() {
     m_replacement = PageId::None;
 
     auto *context = SectionManager::Instance()->globalContext();
-    context->_74 = 2;
+    context->m_vehicleRestriction = GlobalContext::VehicleRestriction::All;
```
Comment (stblr) [inline payload/game/ui/MultiTopPage.cc:67]: Can you also implement the restrictions in multiplayer?

**Review by stblr — CHANGES_REQUESTED** [review]

File: payload/game/ui/SingleTopPage.cc (hunk 2 above)
Comment (GnomedDev) [reply payload/game/ui/SingleTopPage.cc:99]: It only affects CPU drivers, so nothing happens.

**Review by stblr — APPROVED** [review]

**Pull request merged**